legal-api/src/legal_api/services/digital_credentials.py

```diff
@@ -29,131 +30,184 @@ class DigitalCredentialsService:
+            'given_names',
         ],
-        'schema_name': 'business_schema',  # do not change schema name. this is the name registered in aca-py agent
-        'schema_version': '1.0.0'  # if attributes changes update schema_version to re-register
+        # do not change schema name. this is the name registered in aca-py agent
+        'schema_name': 'digital_business_card',
+        # if attributes change update schema_version to re-register
+        'schema_version': '1.0.0'
     }
 
     def __init__(self):
         """Initialize this object."""
         self.app = None
 
         self.api_url = None
-        self.api_key = None
-        self.entity_did = None
+        self.api_token = None
+        self.public_schema_did = None
+        self.public_issuer_did = None
+
+        self.business_schema_name = None
+        self.business_schema_version = None
+        self.business_schema_id = None
+        self.business_cred_def_id = None
 
     def init_app(self, app):
         """Initialize digital credentials using aca-py agent."""
         self.app = app
 
-        self.api_url = app.config.get('ACA_PY_ADMIN_API_URL')
-        self.api_key = app.config.get('ACA_PY_ADMIN_API_KEY')
-        self.entity_did = app.config.get('ACA_PY_ENTITY_DID')
+        self.api_url = app.config.get('TRACTION_API_URL')
+        self.public_schema_did = app.config.get('TRACTION_PUBLIC_SCHEMA_DID')
+        self.public_issuer_did = app.config.get('TRACTION_PUBLIC_ISSUER_DID')
+
+        self.business_schema_name = app.config.get('BUSINESS_SCHEMA_NAME')
+        self.business_schema_version = app.config.get('BUSINESS_SCHEMA_VERSION')
+        self.business_schema_id = app.config.get('BUSINESS_SCHEMA_ID')
+        self.business_cred_def_id = app.config.get('BUSINESS_CRED_DEF_ID')
+
         with suppress(Exception):
-            self._register_business()
-
-    def _register_business(self):
-        """Register business schema and credential definition."""
-        # check for the current schema definition.
-        definition = DCDefinition.find_by(
-            credential_type=DCDefinition.CredentialType.business,
-            schema_name=self.business_schema['schema_name'],
-            schema_version=self.business_schema['schema_version']
-        )
-
-        if definition:
-            if definition.is_deleted:
-                raise Exception('Digital Credentials: business_schema is marked as delete, fix it.')  # noqa: E501; pylint: disable=broad-exception-raised, line-too-long
-        else:
-            # deactivate any existing schema definition before registering new one
-            DCDefinition.deactivate(DCDefinition.CredentialType.business)
-
-            schema_id = self._register_schema(self.business_schema)
-            definition = DCDefinition(
+            self._register_business_definition()
+
+    def _register_business_definition(self):
+        """Fetch schema and credential definition and save a Business definition."""
```

Comment:
so just confirming that this updated function for registering the business definition is good with multiple legal api pods spinning and down throughout any given day.

Reply:
Good question. The logic in there will delete a definition from the db, if found, and will register a new one. Now the logic was there from before so I had assumed there was a good reason for it. But now I question why we need to delete a definition if it exists already? Every time a new pod comes up, the db table would grow perhaps unnecessarily (that's assuming pods share the same persistent volume). All that `init` function is doing is really checking if the Schema and Cred Def exist in the Traction tenant and on the assocaited ledgers. We're already passing all the values from the config needed to save the definition.

Reply:
From looking at the previous logic, it didn't actually deactivate an active definition if it existed.  It only deactivated all definitions of credential_type = business if the definition couldn't be found.

So I think deactivating the an active definition on every app initialization is not what we want?

Reply:
Perhaps I'm reading that wrong but it looks like it's querying for a definition with type along with the schema name and version of interest and deactivating it regardless before registering a new one, which is why I was confused when trying to refactor that code.

>So I think deactivating an active definition on every app initialization is not what we want? Like you said, the definition table would grow pretty quickly. And the main thing is there is no reason to delete the existing definition.

Agreed, I think it makes more sense to maybe do the sanity check against Traction and return the existing definition and only create a new one if its not found (which would be the case for the first pod and anytime the schema id and cred def change)

Reply:
so this is how I interpreted the original code.  Also, I checked the definition table in dev/test and there was only definition entry in each env.

![image](https://github.com/bcgov/lear/assets/6685223/a32fa31c-6ed0-4abb-8d71-d817404364b0)

Reply:
Right. The indentation...Gotcha 👍